manifests/renew.pp

```diff
@@ -86,13 +90,37 @@
   ]).filter | $arg | { $arg =~ NotUndef and $arg != [] }
   $command = join($_command, ' ')
 
-  cron { 'letsencrypt-renew':
-    ensure   => $cron_ensure,
-    command  => $command,
-    user     => 'root',
-    hour     => $cron_hour,
-    minute   => $cron_minute,
-    monthday => $cron_monthday,
+  if timer_type == 'cron' {
+    cron { 'letsencrypt-renew':
+      ensure   => $cron_ensure,
+      command  => $command,
+      user     => 'root',
+      hour     => $cron_hour,
+      minute   => $cron_minute,
+      monthday => $cron_monthday,
+    }
+  } elsif $timer_type == 'systemd' {
+    systemd::unit_file { 'letsencrypt.timer':
```

Comment:
Before using Systemd, we should check that Systemd is the init system used on the node. For this, the module [camptocamp/systemd custom fact](https://github.com/camptocamp/puppet-systemd/blob/master/lib/facter/systemd.rb) can be used.